stratum/hal/lib/barefoot/bf_sde_wrapper.cc

```diff
@@ -562,15 +567,68 @@ ::util::Status TableData::SetActionMemberId(uint64 action_member_id) {
 }
 
 ::util::Status TableData::GetActionMemberId(uint64* action_member_id) const {
-  return GetField(*table_data_, "$ACTION_MEMBER_ID", action_member_id);
+  const bfrt::BfRtTable* table;
+  RETURN_IF_BFRT_ERROR(table_data_->getParent(&table));
+  bf_rt_id_t field_id;
+  bfrt::DataType data_type;
+  if (table->actionIdApplicable()) {
+    bf_rt_id_t action_id;
+    RETURN_IF_BFRT_ERROR(table_data_->actionIdGet(&action_id));
+    RETURN_IF_BFRT_ERROR_WITHOUT_LOGGING(
+        table->dataFieldIdGet("$ACTION_MEMBER_ID", action_id, &field_id));
```

Comment:
Maybe I understand incorrectly
Is it possible to combine an action + a member?
In P4Runtime spec, we cannot combine action + action member in a table entry

Reply:
I don't think so. Maybe we can use `actionIdApplicable()` has the first check.

Reply:
I think the `is_active` below will be false if the table uses direct action(normal action) instead of indirect action(action member)

Reply:
But we can still check by using `actionIdApplicable` and return earlier

Reply:
> But we can still check by using `actionIdApplicable` and return earlier

See latest change. This seems to work just fine.